nbclient/client.py

```diff
@@ -661,13 +691,14 @@ def _passed_deadline(self, deadline):
             return True
         return False
 
-    def _check_raise_for_error(self, cell, exec_reply):
+    def _check_raise_for_error(self, cell, cell_index, exec_reply):
         cell_allows_errors = self.allow_errors or "raises-exception" in cell.metadata.get(
             "tags", []
         )
 
-        if self.force_raise_errors or not cell_allows_errors:
-            if (exec_reply is not None) and exec_reply['content']['status'] == 'error':
+        if (exec_reply is not None) and exec_reply['content']['status'] == 'error':
+            run_hook(self.on_cell_error, cell, cell_index)
```

Comment:
Do you want this hook for any error (including ignored ones)? It might require that it be specified that suppressed errors would trigger the error handling hook. The caller below may not know if it's a suppressed error or not

Reply:
Yeah I explicitly did it this way since people wouldn't use the hook if they were also suppressing errors. What kind of errors make it here that wouldn't be suppressed?